forc-plugins/forc-client/src/util/gas.rs

```diff
@@ -17,6 +17,7 @@ pub fn get_gas_limit(gas: &Gas, chain_info: ChainInfo) -> u64 {
     if let Some(gas_limit) = gas.limit {
         gas_limit
     } else {
-        chain_info.consensus_parameters.max_gas_per_tx
+        //TODO: @hal3e estimate real `gas_used`
```

Comment:
Setting the script gas limit to `max_gas_per_tx` leads to a validation error. In the rust-sdk we dry run the `tx` first to get `gas_used` and set this as the limit. @xgreenx what do you recommend we do here ?

Reply:
@hal3e Yeah, I was running into the same error on the TS SDK. 

Then I started doing exactly like you; a `dryRun` call then I get the `gas_used` and set it to `gas_limit`. 

However the `gas_limit` needs to be set on a `dryRun` call as well. I wonder what is the correct max value.

In the meanwhile I am using `max_gas_per_tx / 2`

Reply:
Why can't we reuse the logic from Rust SDK to estimate/send transactions?

Reply:
In this case we can as we have a provider. I have updated the code.